protocol/communities/manager.go

```diff
@@ -1038,148 +1082,214 @@ func (m *Manager) reevaluateMembers(communityID types.HexBytes) (*Community, map
 			continue
 		}
 
-		isCurrentRoleTokenMaster := community.IsMemberTokenMaster(memberPubKey)
-		isCurrentRoleAdmin := community.IsMemberAdmin(memberPubKey)
+		revealedAccount, memberHasWallet := membersAccounts[memberKey]
+		if !memberHasWallet {
+			result.membersToRemove[memberKey] = struct{}{}
+			continue
+		}
 
-		revealedAccount, exists := membersAccounts[memberKey]
-		memberHasWallet := exists
+		accountsAndChainIDs := revealedAccountsToAccountsAndChainIDsCombination(revealedAccount)
+
+		result.membersRoles[memberKey] = &reevaluateMemberRole{
+			old: community.MemberRole(memberPubKey),
+			new: protobuf.CommunityMember_ROLE_NONE,
```

Comment:
Should we add a comment here stating that this is the temporary state?

Reply:
If possible, I would try to avoid commenting on this. The logic is pretty straightforward, and it is not a temporary state in all cases. Once the reevaluation starts, it is assumed the member has no role. For instance, if admin permission is removed from an open community, this will not be overridden and will be the final state.